compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -705,24 +710,40 @@ impl<'tcx> interpret::Machine<'tcx> for CompileTimeMachine<'tcx> {
         interp_ok(())
     }
 
-    fn before_alloc_read(ecx: &InterpCx<'tcx, Self>, alloc_id: AllocId) -> InterpResult<'tcx> {
+    fn before_alloc_access(
+        tcx: TyCtxtAt<'tcx>,
+        machine: &Self,
+        alloc_id: AllocId,
+    ) -> InterpResult<'tcx> {
+        if machine.last_frame_popped {
```

Comment:
you can just check `machine.stack().is_empty()`, instead of tracking `last_frame_popped` separately.